### ISSUE TITLE
[debian8] Cannot restore snapshot

### ISSUE THREAD

**Issue by eligao** [gh2o/digitalocean-debian-to-arch#42]
The script of branch `debian8` seems to be still incompatible with snapshots, similar to the issue stated on #7 
When trying to create a new droplet from an existing arch snapshot converted from debian8, it always fails with 

> droplet create event failed please try again

I ticketed DigitalOcean, they said:

> I see that those create attempts are running into an error with the message "Failed to configure network interfaces", which can happen if you're trying to use custom operating systems like that. It seems /etc/network/interfaces is pointing somewhere outside of the filesystem or to a different partition

But the script from branch  `debian8` seems to have created a `doroot` partition and corresponding `/etc/network/interfaces` in it. Is the partition probably misplaced so that DigitalOcean couldn't find this partition?

**Comment by bulletmark** [comment]
I just tried to restore my debian8->arch snapshot and ran into this issue. I have a ticket outstanding with DO but came here and discovered this unfortunate issue. :(

**Comment by bulletmark** [comment]
I set this server up, made a snapshot, then deleted the droplet. So I guess I am stuffed now?

**Comment by TrueBers** [comment]
Same here as well

**Comment by al-jshen** [comment]
Same here. Contacted them and they said there's nothing they can do to help. Any solutions?

**Comment by bulletmark** [comment]
@AL-JShen there were no answers here or at DO so I moved my server to Vultr who provide the latest Arch install ISO. A few clicks and commands and 10 mins later you have a Arch system up and running.

**Comment by gh2o** [comment]
Hey folks! Sorry for the late response. I did some research on the issue, and it appears that in the snapshot restore process, DO added something that writes to `/etc/udev/rules.d` on the DOROOT partition.

I just pushed a fix to allow snapshots to work again. To update your droplet such that any future snapshots created from it can be restored, run the following commands:

```
mkdir -p /mnt/doroot
mount /dev/disk/by-label/DOROOT /mnt/doroot
mkdir -p /mnt/doroot/etc/udev/{rules,hwdb}.d
umount /mnt/doroot
```

The most unfortunate part about this is that snapshots created prior to creating the `rules.d` directory will not be restoreable without DigitalOcean intervention. Sorry for the inconvenience.
